src/cmake-tools.ts

```diff
@@ -494,24 +495,42 @@ export class CMakeTools implements vscode.Disposable, api.CMakeToolsAPI {
         } else if (result === changeSetting) {
           // Open the search file dialog from the path set by cmake.sourceDirectory or from the current workspace folder
           // if the setting is not defined.
-          const openOpts: vscode.OpenDialogOptions = {
-            canSelectMany: false,
-            defaultUri: vscode.Uri.file(this.folder.uri.fsPath),
-            filters: {"CMake files": ["txt"], "All files": ["*"]},
-            openLabel: "Load",
-          };
-          const cmakeListsFile = await vscode.window.showOpenDialog(openOpts);
-          if (cmakeListsFile) {
-            const fullPathDir: string = path.parse(cmakeListsFile[0].fsPath).dir;
-            const relPathDir: string = lightNormalizePath(path.relative(this.folder.uri.fsPath, fullPathDir));
-            const joinedPath = "${workspaceFolder}/".concat(relPathDir);
-            vscode.workspace.getConfiguration('cmake', this.folder.uri).update("sourceDirectory", joinedPath);
+          interface FileItem extends vscode.QuickPickItem {
+            fullPath: string;
+        }
+          const existingCmakeListsFiles: string[] | undefined = util.getAllFilePaths(this.folder.uri, "CMakeLists.txt");
+          const items: FileItem[] = existingCmakeListsFiles ? existingCmakeListsFiles.map<FileItem>(file => ({
+            label: util.getRelativePath(file, this.folder.uri.fsPath) + "/CMakeLists.txt",
+            fullPath: file
+          })) : [];
+          items.push({ label: localize("browse.for.cmakelists", "[Browse for CMakeLists.txt]") , fullPath: "", description: "Search for CMakeLists.txt on this computer" });
+          const selection: FileItem | undefined = await vscode.window.showQuickPick(items, {
+            placeHolder: (items.length === 1 ? localize("cmakelists.not.found", "No CMakeLists.txt was found.") : localize("select.cmakelists", "Select CMakeLists.txt"))
+          });
+          let selectedFile: string | undefined;
+          if (!selection) {
+            break; // User canceled it.
+          } else if (selection.label == "[Scan for CmakeLists.txt]") {
```

Comment:
>label == "[Scan for CmakeLists.txt] [](http://example.com/codeflow?start=31&length=35)

label is localized and doesn't match this string.  Can you save the localized string in a variable and test against that instead?